spec/features/admin_payroll_runs_spec.rb

```diff
@@ -91,6 +91,45 @@
     expect(page).to have_content payroll_run.claims.count
     expect(page).to have_content "Downloaded No"
     expect(page).to have_field("payroll_run_download_link", with: new_admin_payroll_run_download_url(payroll_run))
+
+    expect(page).to have_content payroll_run.payments[0].id
+    expect(page).to have_content payroll_run.payments[1].id
+
+    expect(page).to have_link(href: remove_admin_payroll_run_payment_path(id: payroll_run.payments[0].id, payroll_run_id: payroll_run.id))
+    expect(page).to have_link(href: remove_admin_payroll_run_payment_path(id: payroll_run.payments[1].id, payroll_run_id: payroll_run.id))
+  end
+
+  scenario "Service operator cannot see removal buttons when confirmation report has been uploaded" do
+    sign_in_to_admin_with_role(DfeSignIn::User::SERVICE_OPERATOR_DFE_SIGN_IN_ROLE_CODE)
+
+    payroll_run = create(:payroll_run, :confirmation_report_uploaded, claims_counts: {MathsAndPhysics => 1, StudentLoans => 1})
+
+    click_on "Payroll"
+    click_on "View #{payroll_run.created_at.strftime("%B")} payroll run"
+
+    expect(page).to have_content payroll_run.payments[0].id
+    expect(page).to have_content payroll_run.payments[1].id
+
+    expect(page).to_not have_css("form[action=\"#{admin_payroll_run_payment_path(payroll_run_id: payroll_run.id, id: payroll_run.payments[0].id)}\"]")
+    expect(page).to_not have_css("form[action=\"#{admin_payroll_run_payment_path(payroll_run_id: payroll_run.id, id: payroll_run.payments[1].id)}\"]")
+  end
+
+  scenario "Service operator can remove a payment from a payroll run" do
+    sign_in_to_admin_with_role(DfeSignIn::User::SERVICE_OPERATOR_DFE_SIGN_IN_ROLE_CODE)
+    payroll_run = create(:payroll_run, claims_counts: {MathsAndPhysics => 1, StudentLoans => 1})
+
+    click_on "Payroll"
+    click_on "View #{payroll_run.created_at.strftime("%B")} payroll run"
+
+    click_on "Remove", match: :first
+
+    expect(page).to have_content("Are you sure you want to remove payment #{payroll_run.payments.first.id} from the payroll confirmation report?")
+
+    expect {
+      click_on "Remove payment"
+    }.to change(payroll_run.reload.payments, :count).by(-1)
+
+    expect(page).to have_content("Payment has been removed from payroll run")
```

Comment:
We should also confirm the specific payment was removed, rather than just that the total count was decreased by one, plus that the specific claim references appear on the page.